src/Symfony/Component/Validator/Constraints/IsinValidator.php

```diff
@@ -71,22 +62,21 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        if (!$this->isCorrectChecksum($value)) {
+        if (!$this->isChecksumCorrect($this->createLuhnInput($value))) {
             $this->context->buildViolation($constraint->message)
                 ->setParameter('{{ value }}', $this->formatValue($value))
                 ->setCode(Isin::INVALID_CHECKSUM_ERROR)
                 ->addViolation();
         }
     }
 
-    private function isCorrectChecksum(string $input): bool
+    private function createLuhnInput(string $input): string
     {
         $characters = str_split($input);
         foreach ($characters as $i => $char) {
             $characters[$i] = \intval($char, 36);
         }
-        $number = implode('', $characters);
 
-        return 0 === $this->validator->validate($number, new Luhn())->count();
```

Comment:
Another solution could be to use `$this->context->getValidator()->validate()` Calling the validator from a ConstraintValidator to apply another constraint is a first-class use class

Reply:
Hmm, the mock setup in that test somehow doesn't allow me to do that. 🤔

Reply:
then fix the mock (or the usage of the mock). See AtLeastOneOfValidator for an existing usage of this pattern, which is tested.

Reply:
Thanks, its works now. This is probably the better way to do this.